### PR TITLE
CAMEL-16537 Allow to define connection via uri params

### PR DESCRIPTION
**Pull request by mmelko**
This enables to define mongodb connection also in uri using `mongodb:db?hosts=host:21017&username=user&password=pwd`. so connection bean isn't mandatory. That can be useful in some scenarios for example to create kamelet bindings.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
LGTM, let's wait for more reviews :-)

File: components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java [hunk 1]
```diff
@@ -68,6 +69,13 @@
     @UriPath(description = "Sets the connection bean reference used to lookup a client for connecting to a database.")
     @Metadata(required = true)
     private String connectionBean;
+
+    @UriParam
```
Comment (davsclaus) [inline components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java:73]: (label = security, secret = true)

File: components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java [hunk 2]
```diff
@@ -68,6 +69,13 @@
...
+    private String username;
+    @UriParam
```
Comment (davsclaus) [inline components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java:75]: (label = security, secret = true)

to have username and password as security options and also as secret so they are not leaked in logs etc

Comment (mmelko) [reply components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java:75]: thanks!

**Review by davsclaus — APPROVED** [review]

**Pull request merged**